gr-azure-software-radio/python/blob_sink.py

```diff
@@ -95,6 +86,55 @@ def __init__(self, np_dtype: np.dtype, vlen: int = 1, authentication_method: str
         self.queue = queue.Queue(maxsize=queue_size)
         self.buf = np.zeros((self.block_len*self.item_size, ), dtype=np.byte)
         self.num_buf_bytes = 0
+        self.log = gr.logger(f"gr_log.{self.symbol_name()}")
+        
+        # If user accidently left .sigmf-data or meta file extension, strip it off
+        if sigmf and '.sigmf' in blob_name:
+            blob_name = blob_name.rsplit('.', 1)[0]
+        
+        if sigmf:
+            meta_blob_client = self.blob_service_client.get_blob_client(container=container_name,
+                                                                        blob=blob_name + '.sigmf-meta')
+            if np_dtype == np.complex64:
+                datatype_str = 'cf32_le'
+            elif np_dtype == np.float32:
+                datatype_str = 'rf32_le'
+            elif np_dtype == np.int32: # TODO: Check that our original usage of np.int32 makes sense; usually complex int16s (iShorts) are saved as int16s not int32s
```

Comment:
In the Blob Sink yaml (i.e. that existed before my changes) I'm not sure we are dealing with i-shorts correctly, it shows it as using int32 datatype, but i-shorts usually use int16 and then you just have to keep track separately that they are interleaved IQ instead of just reals.  That's why when you read in a file that is i-shorts, you have to use the i-short-to-complex block.

Reply:
let's open an issue to investigate

Reply:
I opened an issue and updated the comment to link to the issue (that line will have to be modified once the issue is addressed anyway so we can remove the link once its fixed)